apps/src/storage/levelbuilder/Dataset.jsx

```diff
@@ -14,17 +37,58 @@ class Dataset extends React.Component {
     // it as an array."
     // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
     tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
-      .isRequired
+      .isRequired,
+    onUploadComplete: PropTypes.func.isRequired
```

Comment:
I've always seen props provided from redux marked as required, but is there a reason why it shouldn't be?

Reply:
i don't feel very strongly about it, but wanted to ask because someone else using this component may not want/need an `onUploadComplete` callback. but since it comes from redux, i think what you have makes more sense